src/codeu/chat/client/commandline/Chat.java

```diff
@@ -619,6 +634,56 @@ public void invoke(Tokenizer args) {
       }
     });
 
+    // TOGGLE-PERMISSION (user's permission to be changed) (permission)
+    //
+    // Add a command to allow authorized users to toggle permissions of other users
+    //
+    panel.register("toggle-permission", new Panel.Command() {
+      @Override
+      public void invoke(Tokenizer args) {
+        final String name = args.hasNext() ? args.next().trim() : "";
+
+        int permission = 0;
+
+        while (args.hasNext()) {
+          final String permissionString = args.hasNext() ? args.next().trim().toLowerCase() : "";
+          switch (permissionString) {
+            case "admin":  permission = permission ^ ConversationHeader.ADMIN_PERM;
+                           break;
+            case "owner":  permission = permission ^ ConversationHeader.OWNER_PERM;
+                           break;
+            case "member": permission = permission ^ ConversationHeader.MEMBER_PERM;
+                           break;
+            default:       System.out.println("ERROR: input is not a valid condition"); return;
+          }
+        }
+        if (conversation.conversation.isMember(conversation.user.id)) {
```

Comment:
This client side check isn't necessary.  Actual permissions checks should all happen on the server.